src/core/lambda/io.rs

```diff
@@ -144,6 +160,12 @@ impl<'a, Ctx: ResolverContextLike<'a> + Sync + Send> CacheKey<EvaluationContext<
             IO::Http { req_template, .. } => req_template.cache_key(ctx),
             IO::Grpc { req_template, .. } => req_template.cache_key(ctx),
             IO::GraphQL { req_template, .. } => req_template.cache_key(ctx),
+            IO::Js { name: method } => {
+                // TODO
+                let mut hasher = std::hash::DefaultHasher::new();
+                method.hash(&mut hasher);
+                hasher.finish()
+            }
```

Comment:
Intresting... I think you have hit an edge case.

While you are at it, can you add this documentation:

> Generating `cache_key` allows us to cache IO calls within the context of an GraphQL request. This ensures that if an IO is executed once, we don't execute it again and wait for the initiated request to finish or serve from cache if the data is already available. The cache_key is generated based on what's the IO type AND what's the value in ctx.


Hopefully the documentation above makes sense. The problem in your implementation is that the JS function will be called only once irrespective of what the input is because the JS function takes context as input and you don't use it for hashing.


Assuming that you can only insert one `@js` on one field, the caching doesn't make a lot of sense because the ctx is almost always going to be different. In essense — JS doesn't fit the current requirements of a typical IO, which is that it should be cacheable. 

There are two ways we can go from here — 
1. Move JS out of IO, but that's not right, JS can fail, and in other environments it might actually be an IO.
2. Drop the requirement of being cacheable for IO. That means `cache_key` should return an `Option<u64>` instead.


My inclination is towards the second one. It's possible that there other IO operations that aren't cacheable perhaps.

Phew!

Reply:
resolved in #2009